lib/node_modules/@stdlib/array/ones-like/docs/types/index.d.ts

```diff
@@ -341,356 +81,13 @@ declare function onesLike( x: Float32Array, dtype?: DataType ): Float32Array;
-* // returns <Uint8ClampedArray>[ 1, 1 ]
-*/
-declare function onesLike( x: Uint8ClampedArray, dtype?: DataType ): Uint8ClampedArray;
-
-/**
-* Creates an array filled with ones and having the same length and data type as a provided input array.
-*
-* The function supports the following data types:
-*
-* -   `float64`: double-precision floating-point numbers (IEEE 754)
-* -   `float32`: single-precision floating-point numbers (IEEE 754)
-* -   `complex128`: double-precision complex floating-point numbers
-* -   `complex64`: single-precision complex floating-point numbers
-* -   `int32`: 32-bit two's complement signed integers
-* -   `uint32`: 32-bit unsigned integers
-* -   `int16`: 16-bit two's complement signed integers
-* -   `uint16`: 16-bit unsigned integers
-* -   `int8`: 8-bit two's complement signed integers
-* -   `uint8`: 8-bit unsigned integers
-* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
-* -   `generic`: generic JavaScript values
-*
-* @param x - input array from which to derive the output array length
-* @param dtype - data type
-* @returns filled array
-*
-* @example
-* var zeros = require( '@stdlib/array/zeros' );
-*
-* var x = zeros( 2, 'generic' );
-* // returns [ 0, 0 ]
-*
-* var y = onesLike( x );
-* // returns [ 1, 1 ]
-*/
-declare function onesLike( x: Array<any>, dtype?: DataType ): Array<number>;
-
-/**
-* Creates an array filled with ones and having the same length and data type as a provided input array.
-*
-* The function supports the following data types:
-*
-* -   `float64`: double-precision floating-point numbers (IEEE 754)
-* -   `float32`: single-precision floating-point numbers (IEEE 754)
-* -   `complex128`: double-precision complex floating-point numbers
-* -   `complex64`: single-precision complex floating-point numbers
-* -   `int32`: 32-bit two's complement signed integers
-* -   `uint32`: 32-bit unsigned integers
-* -   `int16`: 16-bit two's complement signed integers
-* -   `uint16`: 16-bit unsigned integers
-* -   `int8`: 8-bit two's complement signed integers
-* -   `uint8`: 8-bit unsigned integers
-* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
-* -   `generic`: generic JavaScript values
-*
-* @param x - input array from which to derive the output array length
-* @param dtype - data type
-* @returns filled array
-*
-* @example
-* var zeros = require( '@stdlib/array/zeros' );
-*
-* var x = zeros( 2, 'float64' );
-* // returns <Float64Array>[ 0.0, 0.0 ]
-*
-* var y = onesLike( x );
-* // returns <Float64Array>[ 1.0, 1.0 ]
-*
-* @example
-* var zeros = require( '@stdlib/array/zeros' );
-*
 * var x = zeros( 2, 'float64' );
 * // returns <Float64Array>[ 0.0, 0.0 ]
 *
 * var y = onesLike( x, 'float32' );
 * // returns <Float32Array>[ 1.0, 1.0 ]
 */
-declare function onesLike( x: AnyArray, dtype?: DataType ): AnyArray;
+declare function onesLike<T extends keyof NumericAndGenericDataTypeMap<number>>( x: AnyArray, dtype: T ): NumericAndGenericDataTypeMap<number>[T];
```

Comment:
This is fine for this PR, but the prior defn does not properly account for accessor arrays.